bpf/lib/fib.h

```diff
@@ -66,8 +66,8 @@ static __always_inline bool fib_ok(int ret)
   */
 static __always_inline int
 fib_do_redirect(struct __ctx_buff *ctx, const bool needs_l2_check,
-		const struct bpf_fib_lookup_padded *fib_params, __s8 *fib_ret,
-		int *oif)
+		const struct bpf_fib_lookup_padded *fib_params,
+		bool lookup_done, __s8 *fib_ret, int *oif)
```

Comment:
Commenting here given GH doesn't allow to comment on the commit message itself. Could you elaborate on the XDP situation? We do install managed neigh entries for all cluster nodes, so that for XDP we have the backends readily available. How does the scenario/issue you are fixing apply there?

Reply:
Hm, I was only concerned about the reply path.

Yes, the forward path doesn't look good either. Say `tail_nodeport_nat_egress_ipv4()` - it calls `fib_redirect()`. But why would it **ever** be allowed to skip the L2 resolution? It won't find the worker node's MAC address in the neighbour map. So this seems like another bug - we need better instruction by the caller whether the FIB lookup is required, and whether a fallback to neighbour map should even be attempted (it doesn't make sense for quite a few paths ...)

Reply:
>Yes, the forward path doesn't look good either. Say tail_nodeport_nat_egress_ipv4() - it calls fib_redirect(). But why would it ever be allowed to skip the L2 resolution? It won't find the worker node's MAC address in the neighbour map. So this seems like another bug - we need better instruction by the caller whether the FIB lookup is required, and whether a fallback to neighbour map should even be attempted (it doesn't make sense for quite a few paths ...)

This is addressed by the second patch now (ensuring that we perform a FIB lookup when the neigh-resolver is not available).